### PR TITLE
Combined pull requests with func tests fixes

### PR DESCRIPTION
**Pull request by ntymtsiv**

### CODE REVIEW

File: devlab/provision/devstack/devstack.sh [hunk 1]
```diff
@@ -45,3 +45,14 @@ function gen_openrc {
 }
 gen_openrc "../openrc_admin" "admin" "admin"
 gen_openrc "../openrc_demo" "demo" "demo"
+
+#clean public and privet networks and router1
```
Comment (plomakin) [inline devlab/provision/devstack/devstack.sh]: privat

File: devlab/tests/config.py [hunk 2]
```diff
@@ -22,9 +22,13 @@
...
 ]
 
+user_tenant_roles = [
+    {'user1': [{'tenant': 'tenant1', 'role': 'SomeRole'}]}
+]
+
 # Roles to create/delete
 roles = [
-    {'name': 'SomeRole'}
+    {'name': 'SomeRole'}, {'name': 'SecondRole'}
```
Comment (plomakin) [inline devlab/tests/config.py]: Is 'SecondRole' used somewhere?

File: devlab/tests/functional_test.py [hunk 3]
```diff
@@ -60,10 +60,15 @@ def filter_networks(self):
     # 'name' parameter in it's detail.
     # Implimentation of Subnet filtering for no name subnets is still needed.
     def filter_subnets(self):
-        subnets = [i['name'] for i in config.subnets]
-        for i in config.tenants:
-            if 'subnets' in i:
-                for subnet in i['subnets']:
+        subnets = [i['name'] for net in config.networks if net.get('subnets')
```
Comment (plomakin) [inline devlab/tests/functional_test.py]: better rename this variable to 'subnet_names'

File: devlab/tests/functional_test.py [hunk 4]
```diff
@@ -60,10 +60,15 @@ def filter_networks(self):
...
+                   for i in net['subnets']]
+        for tenant in config.tenants:
+            if 'networks' not in tenant:
+                continue
+            for network in tenant['networks']:
+                if 'subnets' not in network:
+                    continue
+                for subnet in network['subnets']:
```
Comment (plomakin) [inline devlab/tests/functional_test.py]: The main issues here - subnet name can be empty... So maybe you should think about how to filter using other parameter.

File: devlab/tests/generate_load.py [hunk 5]
```diff
@@ -162,14 +241,28 @@ def switch_user(self, user, password, tenant):
...
+        super(Prerequisites, self).__init__(config, cloud_prefix)
+        # will be filled during create all networking step
+        self.ext_net_id = None
+        # object of Prerequisites for dst cloud
+        self.dst_cloud = None
+        self.clean_tools = CleanEnv(config, cloud_prefix)
+
+    @clean_if_exists
+    def create_users(self, users=None):
         def get_params_for_user_creating(_user):
```
Comment (plomakin) [inline devlab/tests/generate_load.py]: Are you sure it will not be re-used in other methods?

File: devlab/tests/generate_load.py [hunk 6]
```diff
@@ -715,44 +864,105 @@ def wait_until_vms_all_deleted():
...
                                   if tenant.get('networks')])
+        nets_names = [net['name'] for net in nets]
+        for network in self.neutronclient.list_networks()['networks']:
+            if network['name'] not in nets_names:
+                continue
+            self.neutronclient.delete_network(self.get_net_id(network['name']))
+            print('Network "%s" has been deleted' % network['name'])
+
+    def clean_routers(self):
+        def clean_router_ports(router_id):
```
Comment (plomakin) [inline devlab/tests/generate_load.py]: >>Are you sure it will not be re-used in other methods?

File: devlab/tests/test_resource_migration.py [hunk 7]
```diff
@@ -1,11 +1,12 @@
-import re
-import pprint
 import config
-import unittest
-import subprocess
 import functional_test
 
-from time import sleep
+import pprint
```
Comment (plomakin) [inline devlab/tests/test_resource_migration.py]: Is it used somewhere? I believe it lost after debugging.

**Comment by plomakin** [comment]
Overall pull request is good. Please have a look at comments.

File: devlab/tests/generate_load.py (hunk 5 above)
Comment (ntymtsiv) [inline devlab/tests/generate_load.py]: create_users should be used for user creating. I do not see cases, when this method will be re-used in other methods.

File: devlab/tests/config.py (hunk 2 above)
Comment (ntymtsiv) [inline devlab/tests/config.py]: 'SomeRole' is created on dst cloud before migration (this role won't migrate from src) (SAIC-522). SecondRole is used for role migration testing.

File: devlab/tests/test_resource_migration.py (hunk 7 above)
Comment (ntymtsiv) [inline devlab/tests/test_resource_migration.py]: It is used for error message printing:
https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/devlab/tests/test_resource_migration.py#L431

**Comment by roman-verchikov** [comment]
+1, can be merged.

**Pull request merged**